osu.Game/Performance/HighPerformanceSessionManager.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Runtime;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+
+namespace osu.Game.Performance
+{
+    public partial class HighPerformanceSessionManager : Component
+    {
+        private GCLatencyMode originalGCMode;
+
+        public IDisposable BeginSession()
+        {
+            enableHighPerformanceSession();
+            return new InvokeOnDisposal<HighPerformanceSessionManager>(this, static m => m.disableHighPerformanceSession());
+        }
+
+        private void enableHighPerformanceSession()
+        {
+            originalGCMode = GCSettings.LatencyMode;
+            GCSettings.LatencyMode = GCLatencyMode.LowLatency;
+
+            // Without doing this, the new GC mode won't kick in until the next GC, which could be at a more noticeable point in time.
+            GC.Collect(0);
+        }
+
+        private void disableHighPerformanceSession()
+        {
+            if (GCSettings.LatencyMode == GCLatencyMode.LowLatency)
+                GCSettings.LatencyMode = originalGCMode;
+
+            // No GC.Collect() as we were already collecting at a higher frequency in the old mode.
+        }
```

Comment:
I'd like to see logging in these methods too to be able to tell when this is being toggled.